Added default Icon to all JavaFX dialogs that inherit from FXAlert

boceckts opened the pull request:
This will add a small JabRef icon to the upper left of every dialog that is a subclass of FXAlert.
Also the keybindings dialog has used the localization strings from the menu files which displayed an and ("&") sign in front of the actual tranlsation. I changed them to use the "normal" localization files and updated the localization files as well. (There were quite some keys missing and/or not available)

This is the current state of the dialog (applies to all JavaFX dialogs)
![keybindingsnoicon](https://cloud.githubusercontent.com/assets/15340757/18353234/8c3f0c84-75e1-11e6-8bf7-ca62184db0ee.PNG)

and after the change
![keybinding](https://cloud.githubusercontent.com/assets/15340757/18353262/9fba8ae0-75e1-11e6-9b64-268c4fa40208.PNG)
![keybiningssave](https://cloud.githubusercontent.com/assets/15340757/18353266/a31e5cd4-75e1-11e6-96ff-3f2dd42c6c0b.PNG)
![journalduplicate](https://cloud.githubusercontent.com/assets/15340757/18353270/a46ca9d8-75e1-11e6-9f34-9d761d38adea.PNG)
![about](https://cloud.githubusercontent.com/assets/15340757/18353271/a55ca096-75e1-11e6-8e0c-b780247c0252.PNG)




- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [x] Screenshots added (for bigger UI changes)
- [x] Manually tested changed features in running JabRef
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

koppor commented on src/main/java/net/sf/jabref/gui/keyboard/KeyBindingCategory.java:
```diff
@@ -5,19 +5,19 @@
 
     FILE(
-            Localization.menuTitle("File")),
+            Localization.lang("File")),
```
Why this change? Isn't the category mirroring the menu structure? Menu strings are translated in the menu translation files, not the normal language files.

boceckts replied:
Yes it is but the translation strings have a leading and ("&") sign and since most of the strings are also in the non menu localization files I changed it.

On Thu, Sep 08, 2016 at 10:30 PM, Oliver Kopp < notifications@github.com [notifications@github.com] > wrote:
In src/main/java/net/sf/jabref/gui/keyboard/KeyBindingCategory.java [https://github.com/JabRef/jabref/pull/1940#discussion_r78082976] :

> @@ -5,19 +5,19 @@
>  public enum KeyBindingCategory {
>  
>      FILE(
> -            Localization.menuTitle("File")),
> +            Localization.lang("File")),


Why this change? Isn't the category mirroring the menu structure? Menu strings are translated in the menu translation files, not the normal language files.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub [https://github.com/JabRef/jabref/pull/1940/files/e5277fac194f6ec75df9d693b4c5845511a679a6#r78082976] , or mute the thread [https://github.com/notifications/unsubscribe-auth/AOoU1Zr8Cm_Ea1R-87FnLK3v3kFPVkaXks5qoHBegaJpZM4J4D-T] .

koppor replied:
I would not require the translators to replace terms twice. Why not just removing the `&` after getting the translation from `menuTitle`?

Reason: You are displaying menu strings. And menu strings are contained in `Menu_X.properties`. It might happen that a menu string is translated differently in JAbRef_*.properties. Using `Menu_*.properties` avoids that issue.

koppor commented on src/main/java/net/sf/jabref/logic/l10n/Localization.java:
```diff
@@ -15,7 +15,6 @@
 
     public static final String RESOURCE_PREFIX = "l10n/JabRef";
     public static final String MENU_RESOURCE_PREFIX = "l10n/Menu";
-    public static final String BIBTEX = "BibTeX";
```
Why did you remove that?

boceckts replied:
It was only used in the viewModel where I changed it to Localization.lang("BibTex").

On Thu, Sep 08, 2016 at 10:30 PM, Oliver Kopp < notifications@github.com [notifications@github.com] > wrote:
In src/main/java/net/sf/jabref/logic/l10n/Localization.java [https://github.com/JabRef/jabref/pull/1940#discussion_r78083018] :

> @@ -15,7 +15,6 @@
>  
>      public static final String RESOURCE_PREFIX = "l10n/JabRef";
>      public static final String MENU_RESOURCE_PREFIX = "l10n/Menu";
> -    public static final String BIBTEX = "BibTeX";


Why did you remove that?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub [https://github.com/JabRef/jabref/pull/1940/files/e5277fac194f6ec75df9d693b4c5845511a679a6#r78083018] , or mute the thread [https://github.com/notifications/unsubscribe-auth/AOoU1WoyuZFMXSbh5mdWxJ4KqeHLp2gUks5qoHBvgaJpZM4J4D-T] .

koppor replied:
BibTeX is always spelled BibTeX - see https://github.com/koppor/jabref/issues/73 -- BibTeX is a name like "OS/2" or "Linux" or "LaTeX" and must not be translated.

Thus `Localiatzion.lang("BibTex")` is unfortunately wrong in two aspects: 1) casing, 2) requiring translation

koppor commented on src/main/resources/l10n/JabRef_da.properties:
```diff
@@ -1800,3 +1791,6 @@ should_have_the_first_letter_capitalized=
 shared=
+
+Tools=
+BibTex=
```
BibTeX is always spelled BibTeX - see https://github.com/koppor/jabref/issues/73

BibTeX is a name like "OS/2" or "Linux" or "LaTeX" and must not be translated.

koppor commented on src/main/resources/l10n/JabRef_da.properties:
```diff
@@ -1800,3 +1791,6 @@ should_have_the_first_letter_capitalized=
 
 shared=
+
+Tools=
```
This is a menu title? This, it does not beed to be translated.

koppor commented:
The screenshots look good, but I wonder why you touched the language files. Please do one thing at one PR and do not mix issues in a PR.

boceckts commented:
Since I added something to the non menu translation files I updated them via gradle, that's why there is so many changes. I will change it back to use the menu files.

boceckts commented:
Comments have been addressed

tobiasdiez commented:
LGTM :+1:

tobiasdiez commented on src/main/java/net/sf/jabref/gui/keyboard/KeyBindingCategory.java:
```diff
@@ -23,7 +23,7 @@
 
     private KeyBindingCategory(String name) {
-        this.name = name;
+        this.name = name.replace("&", "");
```
I know Olly made you change this, but actually your previous fix was better.
The KeyBindingCategory is used as the header in the keybinding dialog and thus has absolutely nothing to do with the menu. That the strings exists in the menu is pure (kind of) coincidence. Thus it should be indeed `FILE(Localization.lang("File"))` in `KeyBindingCategory` instead of using `menuTitle` there.

boceckts replied:
Well it's not really coincidence that the categories match the menu titles as I used the menu titles where the shortkey's action is located as the its category.

tobiasdiez replied:
Yes I know, but one may add keyboard shortcuts for actions which are not present in the menu or reorder the categories a bit.

boceckts replied:
Ok then I change it back to use the normal localization files.

koppor replied:
> The KeyBindingCategory is used as the header in the keybinding dialog and
thus has absolutely nothing to do with the menu.

Sure? I thought, this was intentional to mirror the menu structure here...
And thus reuse the menu translations?

boceckts commented:
Comments are addressed.

tobiasdiez commented:
Thanks
